pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -304,6 +312,64 @@ synchronized void runIncrementalPipeline()
     }
   }
 
+  /**
+   * This method periodically checks on Thrift server output directory and commands to create
+   * resource tables. The very first run happens over a minute later once controller spins up. And
+   * it's programmed to check on Thrift server every hour in case it went down for unfortunate
+   * reasons.
+   *
+   * @throws IOException
+   * @throws SQLException
+   */
+  @Scheduled(initialDelay = 60000, fixedDelay = 3600000)
+  private void createResourceTablesOnStart() throws IOException, SQLException {
+    if (!dataProperties.isCreateHiveResourceTables()) {
+      return;
+    }
+
+    DatabaseConfiguration dbConfig =
+        DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig());
+
+    HiveTableManager hiveTableManager =
+        new HiveTableManager(
+            dbConfig.makeJdbsUrlFromConfig(),
+            dbConfig.getDatabaseUser(),
+            dbConfig.getDatabasePassword());
+
+    // Traverse through directories in parquet output directory to collect their names.
+    List<Path> paths;
+    try (Stream<Path> walk =
+        Files.walk(Paths.get(Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX))) {
```

Comment:
This is actually a good question but I don't think we can use `java.nio.file.*`. @chandrashekar-s when you did your changes to enable file access (and tested with GCS), did you also try querying those GCS files from the thrift-server? If not, we should do that and see how that flow works. Then we can answer this question.

Reply:
I am still unclear on how we cannot use `java.nio.file.*`. 
What are the potential gains by using Beam packages here? Are we saying that we are going to get performance improvements cause this is just scan of directories.

The other approach where we are saying that we have parquet files stored in external distributed file system and we mount the remote location to local directory, even in this case, `java.nio.file.*` will come handy.

Is there any potential disadvantage of using this?

Reply:
I realised that the controller application has direct access to the file system either local or remote (unlike the thrift server which uses the mounted file system). Since this piece of code is invoked by the controller app, to make sure this code works correctly irrespective of the file system used, using the Beam FileSystem APIs is the right way.

On the contrary, currently the thrift server talks to the locally mounted file system. Looks like there are options available for mounting the external remote file systems to local file systems, I will explore more on this and get back on the results.

Reply:
Also the root directory which the controller app needs to scan is dynamic and is dependent on dataProperties.getDwhRootPrefix() value, we should avoid scanning the /dwh directly by the controller app.

Reply:
So yeah, the reason behind using Beam files lib in the controller is not performance but the ability of working with GCS (and supposedly other distributed file systems). So given that in PR #524 we moved away from Java IO library, in particular [changes to PipelineManager.java](https://github.com/google/fhir-data-pipes/pull/524/files#diff-ca9fd6fc1db26b0b72097b3561e2ee91bc5b7afbb4fd0842b9c6e38d4b55e6dd), it would be nice if we don't introduce it back.

The question still remains on how we use those files in Spark+Thriftserver, but regardless, if it is not too much work, let's stick with Beam files lib.

Reply:
Ok, I have implemented Beam file based mechanism.